Feat/verbose filter

meditationduck opened the pull request.

meditationduck committed:
✨ Add verbose output filtering and enhance logging in CLI and ClaudeCodeSession

- Introduced a new `--verbose-filter` option in the CLI to filter verbose output by tool names.
- Enhanced logging in `ClaudeCodeSession` to conditionally display tool usage and results based on the specified filters.
- Added utility functions in the logging module to manage verbose filters and determine visibility of tools based on these filters.

meditationduck committed:
✅ format for -v

meditationduck committed:
🔧 Refactor verbose output handling and improve console display

- Adjusted console print formatting for better readability.
- Updated verbose level checks to refine output visibility.
- Enhanced error handling during verbose message processing.
- Commented out unnecessary print statements to reduce clutter.

michprev commented on wake_ai/core/claude.py:
```diff
@@ -37,10 +37,49 @@
+    new_string: str
+
+@dataclass
+class Grob:
```
Grep or Glob? 😄

meditationduck replied:
oops, sorry, it is not used now. fixing.

michprev commented on wake_ai/cli.py:
```diff
@@ -256,8 +256,13 @@ def get_class_that_defined_method(meth):
+@click.option(
+    "--verbose-filter",
+    type=str,
+    help="Filter verbose output by tool names (comma-separated, e.g., 'Bash,Edit,mcp__wake,TodoWrite,Read,Grep')"
```
is this a finite and complete list? maybe Glob missing

meditationduck replied:
Yes, it is better to use a reference. fixing to add all from here
https://docs.anthropic.com/en/docs/claude-code/settings#tools-available-to-claude

meditationduck committed:
🔧 Update verbose filter help text and remove unused Grob class

- Enhanced the help text for the `--verbose-filter` option in the CLI to include additional tool names for better clarity.
- Removed the unused `Grob` class from the `claude.py` file to clean up the codebase.
